src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -213,18 +213,8 @@ async Task<string> IDurableOrchestrationClient.StartNewAsync<T>(string orchestra
 
         private OrchestrationStatus[] GetStatusesNotToOverride()
         {
-            var overridableStates = this.durableTaskOptions.OverridableExistingInstanceStates;
-            if (overridableStates == OverridableStates.NonRunningStates)
-            {
-                return new OrchestrationStatus[]
-                {
-                    OrchestrationStatus.Running,
-                    OrchestrationStatus.ContinuedAsNew,
-                    OrchestrationStatus.Pending,
-                };
-            }
-
-            return new OrchestrationStatus[0];
+            OverridableStates overridableStates = this.durableTaskOptions.OverridableExistingInstanceStates;
+            return overridableStates.ToDedupeStatuses();
```

Comment:
I'm confused. The method name is `GetStatusesNotToOverride`, but the value we're returning seems to be the opposite of this: `overridableStates`. Why is this logic correct? Assuming it is correct, can we rename something to make it _obviously_ correct?